codegen/src/main/ops/org/nd4j/codegen/ops/SDBaseOps.kt

```diff
@@ -0,0 +1,1806 @@
+                with the reduced dimensions having size 1. This can be useful for later broadcast operations (such as subtracting
+                the mean along a dimension).
+                Example: if input has shape [a,b,c] and dimensions=[1] then output has shape:
+                keepDims = true: [a,1,c]
+                keepDims = false: [a,c]
+            """.trimIndent()
+        }
+    }
+
+    Op("normmax") {
+        javaPackage = "org.nd4j.linalg.api.ops.impl.reduce.floating"
+        javaOpClass = "NormMax"
+        legacy = true
+        Input(NUMERIC, "x") { description = "Input variable" }
+        Arg(INT, "dimensions") { count = AtLeast(1); description = "dimensions to reduce over" }
+        Output(NUMERIC, "output"){ description = "Output variable" }
+        Doc(Language.ANY, DocScope.ALL){
+            """
+                Max norm (infinity norm) reduction operation: The output contains the max norm for each tensor/subset along the
+                specified dimensions
+            """.trimIndent()
+        }
+    }
+
+    Op("normmax") {
+        javaPackage = "org.nd4j.linalg.api.ops.impl.reduce.floating"
+        javaOpClass = "NormMax"
+        legacy = true
+        Input(NUMERIC, "x") { description = "Input variable" }
+        Arg(BOOL, "keepDims") { description = "If true: keep the dimensions that are reduced on (as size 1). False: remove the reduction dimensions" }
+        Arg(INT, "dimensions") { count = AtLeast(1); description = "dimensions to reduce over" }
+        Output(NUMERIC, "output"){ description = "Output variable" }
+        Doc(Language.ANY, DocScope.ALL){
+            """
+                Max norm (infinity norm) reduction operation: The output contains the max norm for each tensor/subset along the
+                specified dimensions:
+                out = max(abs(x[i]))
+                Note that if keepDims = true, the output variable has the same rank as the input variable,
+                with the reduced dimensions having size 1. This can be useful for later broadcast operations (such as subtracting
+                the mean along a dimension).
+                Example: if input has shape [a,b,c] and dimensions=[1] then output has shape:
+                keepDims = true: [a,1,c]
+                keepDims = false: [a,c]
+            """.trimIndent()
+        }
+    }
+
+    Op("oneHot") {
+        javaPackage = "org.nd4j.linalg.api.ops.impl.shape"
+        Input(NUMERIC, "indices") { description = "Indices - value 0 to depth-1" }
+        Arg(INT, "depth") { description = "Number of classes" }
+        Arg(INT, "axis") { description = "" }
+        Arg(NUMERIC, "on") { description = "" }
+        Arg(NUMERIC, "off") { description = "" }
+        Output(NUMERIC, "output"){ description = "Output variable" }
+        Doc(Language.ANY, DocScope.ALL){
+            """
+                Convert the array to a one-hot array with walues {@code on} and {@code off} for each entry
```

Comment:
Javadoc specific annotation here - `{@code ... }`
that won't work for python, HTML, etc. @treo - do we need some sort of annotation/convention for this that will work for any generation target (javadoc, c++, python, html/markdown reference docs etc)?

Reply:
Removed all @code tags.